### PR TITLE
3864 lemlist

### PR DESCRIPTION
**Pull request by luancazarine**
Triggers
  - New Activity
  - Unsubscribed Recipient
 
Actions
  - Mark a Lead From All Campaigns as Interested
  - Mark a Lead From One Campaign as Interested
  - Add Lead to Campaign
  - Mark a Lead From All Campaigns as Not Interested
  - Mark a Lead From One Campaign as Not Interested
  - Pause Lead From All Campaigns
  - Remove Lead From Campaign
  - Remove Lead From Unsubscribe
  - Resume Lead From All Campaigns
  - Stop Emailing a Lead in a Campaign
  - Add Recipient to Unsubscribe List
  - Update Lead From Campaign
  - Find
x Create Lead
x Search Lead

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4002"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/lemlist/actions/unsubscribe-lead-from-a-campaign/unsubscribe-lead-from-a-campaign.mjs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    },
+    email: {
+      propDefinition: [
+        lemlist,
+        "email",
+        (c) => ({
+          campaignId: c.campaignId.value,
+        }),
+      ],
+      async options({ campaignId }) {
```
Comment (vellames) [inline components/lemlist/actions/unsubscribe-lead-from-a-campaign/unsubscribe-lead-from-a-campaign.mjs:26]: Why change the options here? Can we do it on `app`?

File: components/lemlist/actions/mark-lead-from-one-campaigns-as-notinterested/mark-lead-from-one-campaigns-as-notinterested.mjs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    email: {
+      propDefinition: [
+        lemlist,
+        "email",
+        (c) => ({
+          campaignId: c.campaignId.value,
+        }),
+      ],
+      async options({ campaignId }) {
```
Comment (vellames) [inline components/lemlist/actions/mark-lead-from-one-campaigns-as-notinterested/mark-lead-from-one-campaigns-as-notinterested.mjs:26]: Why change the options here? Can we do it on `app`?

File: components/lemlist/actions/update-lead-in-a-campaign/update-lead-in-a-campaign.mjs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    },
+    email: {
+      propDefinition: [
+        lemlist,
+        "email",
+        (c) => ({
+          campaignId: c.campaignId.value,
+        }),
+      ],
+      async options({ campaignId }) {
```
Comment (vellames) [inline components/lemlist/actions/update-lead-in-a-campaign/update-lead-in-a-campaign.mjs:26]: Why change the options here? Can we do it on `app`?

File: components/lemlist/lemlist.app.mjs [hunk 4]
```diff
@@ -1,11 +1,223 @@
...
   type: "app",
   app: "lemlist",
-  propDefinitions: {},
+  propDefinitions: {
+    campaignId: {
+      type: "string",
+      label: "Campaign Id",
+      description: "The ID of the campaign.",
+      async options({ page }) {
+        const results = await this.listCampaigns({
```
Comment (vellames) [inline components/lemlist/lemlist.app.mjs:12]: Why change the options here? Can we do it on `app`?

**Review by vellames — CHANGES_REQUESTED** [review]
Just some changes

**Review by vellames — DISMISSED** [review]

**Comment by DilanAthukorala** [comment]
@luancazarine Observing following issues

1. For actions - **delete_lead_from_campaign** , **mark_lead_from_one_campaigns_as_interested** , **mark_lead_from_one_campaigns_as_notinterested** , **unsubscribe_lead_from_a_campaign** , **update_lead_in_a_campaign** 

- Unable to delete a lead, observing - (404 - Not Found)

image




2. For **New_Activity** and **New_Unsubscribed** Sources - Missing Auth Params in the UI

image
image

**Comment by luancazarine** [comment]
@DilanAthukorala I'm sorry, I don't know what happened with your source tests, but they actually have auth params
image
image

**Comment by DilanAthukorala** [comment]
@luancazarine Issue fixes are verified and one minor issue observed in update_lead_in_a_campaign action where value for param "IceBreaker" is not getting updated in lemlist.

File: components/lemlist/actions/mark-lead-from-all-campaigns-as-notinterested/mark-lead-from-all-campaigns-as-notinterested.mjs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import lemlist from "../../lemlist.app.mjs";
+
+export default {
+  key: "lemlist-mark-lead-from-all-campaigns-as-notinterested",
+  name: "Mark Lead From All Campaigns As Not Interested",
+  description: "This action marks a specific lead as not interested using its email in all campaigns. [See the docs here](https://developer.lemlist.com/#mark-as-interested-a-specific-lead-by-email)",
```
Comment (feyzullah) [inline components/lemlist/actions/mark-lead-from-all-campaigns-as-notinterested/mark-lead-from-all-campaigns-as-notinterested.mjs:6]: This link might not be correct
```suggestion
  description: "This action marks a specific lead as not interested using its email in all campaigns. [See the docs here](https://developer.lemlist.com/#mark-as-not-interested-a-specific-lead-by-email)",
```

**Review by feyzullah — CHANGES_REQUESTED** [review]
Hi @luancazarine I added just a few comments. And can you please add a `package.json`?

**Comment by feyzullah** [comment]
Hi @luancazarine is this ready for PR review?

**Review by feyzullah — DISMISSED** [review]

**Comment by DilanAthukorala** [comment]
@luancazarine After the changes there's an issue with listing the email addresses of the leads
image

**Comment by DilanAthukorala** [comment]
Verification completed, This is ready for release.

**Review by feyzullah — APPROVED** [review]

**Pull request merged**